apps/condo/domains/meter/schema/RegisterMetersReadingsService.js

```diff
@@ -342,13 +342,24 @@ const RegisterMetersReadingsService = new GQLCustomSchema('RegisterMetersReading
                         ))
                         continue
                     }
-
-                    if (foundMeter) {
-                        meterId = foundMeter.id
-                    } else {
-                        try {
+                    try {
+                        if (foundMeter) {
+                            meterId = foundMeter.id
+                            await Meter.update(context, foundMeter.id, {
```

Comment:
Looks like it will update meter every time. If integration sends data every day - then lot's of history records will be created

Reply:
Good comment, thanks!
Fixed. Also the test was added.